web/thesauruses/c/17/strings.json

```diff
@@ -0,0 +1,353 @@
+      "name": "Is string all alphanumeric characters?"
+    },
+    "is_decimal": {
+      "not-implemented":"true",
+      "name": "Is string a decimal number?"
+    },
+    "is_all_whitespaces": {
+      "not-implemented":"true",
+      "name": "Is string all whitespace characters?"
+    },
+    "is_all_uppercase": {
+      "not-implemented":"true",
+      "comment": [
+        "//indirect implementation",
+        "//#include<ctype.h>",
+        "char str[100];",
+        "gets(str);",
+        "for(int i=0;i<strlen(str);i++)",
+        " if(isupper(str[i]))",
+            "count+=1;",
+        "if(count==strlen(str)) printf(\"all uppercase characters\");"
+      ],
+      "name": "Is string all uppercase characters?"
+    },
+    "is_all_lowercase": {
+      "not-implemented":"true",
+      "comment": [
+        "//indirect implementation",
+        "//#include<ctype.h>",
+        "char str[100];",
+        "gets(str);",
+        "for(int i=0;i<strlen(str);i++)",
+        " if(islower(str[i]))", 
+            "count+=1;",
+        "if(count==strlen) printf(\"all lowercase characters\");"
+      ],
+      "name": "Is string all lowercase characters?"
+    },
+    "is_in_titlecase": {
+      "not-implemented":"true",
+      "name": "Is string formatted in title case?"
+    },
+    "does_substring_exist": {
+      "not-implemented": "true",
+      "name": "Does a substring exist in a string?"
+    },
+    "find_start_index_of_substring": {
+      "not-implemented": "true",
+      "name": "Find index of where a substring starts"
+    },
+    "find_start_index_of_additional_substring": {
+      "not-implemented": "true",
+      "name": "Find index of an additional substring (or starting at another index)"
+    },
+    "find_start_index_of_substring_from_end": {
+      "not-implemented": "true",
+      "name": "Find substring index starting at end"
+    },
+    "count_occurrences_of_substring": {
+      "not-implemented": "true",
+      "name": "Find number of occurences of a substring"
+    },
+    "get_leftmost_characters": {
+      "not-implemented": "true",
+      "name": "Get a specified number of characters from the left"
+    },
+    "get_rightmost_characters": {
+      "not-implemented": "true",
+      "name": "Get a specified number of characters from the right"
+    },
+    "get_substring_from_start_and_end_index": {
+      "not-implemented": "true",
+      "name": "Return a substring from a string based on starting and ending indices"
+    },
+    "get_substring_from_start_index_and_length": {
+      "not-implemented": "true",
+      "name": "Return a substring from a string based on starting index and size of substring"
+    },
+    "convert_to_uppercase": {
+      "not-implemented":"true",
+      "comment": [
+        "//indirect implementation",
+        "//#include<ctype.h>",
+        "char str[100];",
+        "gets(str);",
+        "for(int i=0;i<strlen(str);i++)",
+        " toupper(str[i])",
+        "puts(str);"
+      ],
```

Comment:
This is implemented, it's `toupper(str[i])`. Just make the `#include` and the `toupper` the code part.